service-python/assessclaimdc7101/src/lib/bp_calculator.py

```diff
@@ -52,14 +53,20 @@ def bp_reader(request_body):
             not_sortable_bp.append(reading)
             reading["dateFormatted"] = ''
             continue  # If there is no date associated
+
+        if reading["systolic"]["value"] == 0 or reading["diastolic"]["value"] == 0:
+            if bp_reading_date >= date_of_claim_date - relativedelta(years=2):
+                partial_bp_two_years.append(reading)  # to be displayed in PDF
+            continue
+
         if bp_reading_date >= date_of_claim_date - relativedelta(years=1):
             bp_reading_in_past_year.append(reading)
         if bp_reading_date >= date_of_claim_date - relativedelta(years=2):
             bp_readings_in_past_two_years.append(reading)
             if reading["systolic"]["value"] >= 160 and reading["diastolic"]["value"] >= 100:
                 elevated_bp_in_past_two_years.append(reading)
 
-    result = {"twoYearsBp": sort_bp(bp_readings_in_past_two_years),
+    result = {"twoYearsBp": sort_bp(bp_readings_in_past_two_years + partial_bp_two_years),
```

Comment:
The partial readings will be added to "twoYearsBp" list to be displayed on the pdf. "twoYearsBpCount" is used for determining sufficiency and will not include partial readings.